trough/sync.py

```diff
@@ -1119,7 +1118,12 @@ def do_segment_promotion(self, segment):
                     temp_file.name)
             source = sqlite3.connect(segment.local_path())
             dest = sqlite3.connect(temp_file.name)
-            sqlitebck.copy(source, dest)
+            # originally: sqlitebck.copy(source, dest)
+            # see https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.backup
+            # may need to update pages value: "the number of pages to copy at a time
+            # If equal to or less than 0, the entire database is copied in a single step.
+            # Defaults to -1."
+            source.backup(dest, pages=1)
```

Comment:
Looks like pages=0 matches sqlitebck.copy's default behavior: https://github.com/husio/python-sqlite3-backup/blob/master/src/sqlitebck.c#L101

Reply:
thanks, @avdempsey
I've updated, to match original code.
I do wonder whether setting a non-default pages value here might be helpful. some trough segments, particularly video segments, per account, become large enough to cause trouble.